src/core/ext/transport/chaotic_good/client_transport.cc

```diff
@@ -0,0 +1,120 @@
+// Copyright 2022 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <grpc/support/port_platform.h>
+
+#include "src/core/ext/transport/chaotic_good/client_transport.h"
+
+#include <stdlib.h>
+
+#include <memory>
+#include <string>
+#include <tuple>
+
+#include "absl/status/statusor.h"
+
+#include <grpc/event_engine/event_engine.h>
+#include <grpc/slice.h>
+#include <grpc/support/log.h>
+
+#include "src/core/ext/transport/chaotic_good/frame.h"
+#include "src/core/ext/transport/chaotic_good/frame_header.h"
+#include "src/core/ext/transport/chttp2/transport/hpack_encoder.h"
+#include "src/core/lib/gprpp/match.h"
+#include "src/core/lib/promise/activity.h"
+#include "src/core/lib/promise/detail/basic_seq.h"
+#include "src/core/lib/promise/event_engine_wakeup_scheduler.h"
+#include "src/core/lib/promise/join.h"
+#include "src/core/lib/promise/loop.h"
+#include "src/core/lib/slice/slice.h"
+#include "src/core/lib/slice/slice_buffer.h"
+#include "src/core/lib/slice/slice_internal.h"
+#include "src/core/lib/transport/promise_endpoint.h"
+
+namespace grpc_core {
+namespace chaotic_good {
+
+ClientTransport::ClientTransport(
+    std::unique_ptr<PromiseEndpoint> control_endpoint,
+    std::unique_ptr<PromiseEndpoint> data_endpoint,
+    std::shared_ptr<grpc_event_engine::experimental::EventEngine> event_engine)
+    : outgoing_frames_(MpscReceiver<ClientFrame>(4)),
+      control_endpoint_(std::move(control_endpoint)),
+      data_endpoint_(std::move(data_endpoint)),
+      control_endpoint_write_buffer_(SliceBuffer()),
+      data_endpoint_write_buffer_(SliceBuffer()),
+      event_engine_(event_engine) {
+  auto hpack_compressor = std::make_shared<HPackCompressor>();
```

Comment:
Lets make hpack_compressor also as a member of the ClientTransport class ? Is there a reason not to do so ?

Reply:
Not really, I think we were using it as a local variable in Constructor because there's no other place need it. But moving to the class member should good, I've updated the code.